### PR TITLE
fix: Background Sender reset on Swoole Fork

### PR DESCRIPTION
**Pull request by PROFeNoM**
### Description

#2636

### Reviewer checklist
- [ ] Test coverage seems ok.
- [ ] Appropriate labels assigned.

### CODE REVIEW

File: ext/ddtrace.stub.php [hunk 1]
```diff
@@ -977,4 +977,9 @@ function dd_untrace(string $functionName, string $className = null): bool {}
      * @param int $timeout Timeout in milliseconds to wait for the flush to complete
      */
     function dd_trace_synchronous_flush(int $timeout): void {}
+
+    /**
+     * @internal
+     */
+    function dd_handle_fork(): void {}
```
Comment (bwoebi) [inline ext/ddtrace.stub.php:984]: Can we please add no new functions to top-level namespace, but add to `DDTrace` or `DDTrace\Internal` as appropriate?

File: ext/ddtrace.c [hunk 2]
```diff
@@ -2136,6 +2136,55 @@ static void dd_ensure_root_span(void) {
     }
 }
 
+/* {{{ proto void dd_handle_fork(): void */
+PHP_FUNCTION(dd_handle_fork) {
+    UNUSED(execute_data);
+    UNUSED(return_value);
+    // CHILD PROCESS
```
Comment (bwoebi) [inline ext/ddtrace.c:2143]: I assume it's still todo to deduplicate with the code in handlers_pcntl.c?

Comment (PROFeNoM) [reply ext/ddtrace.c:2143]: What do you exactly mean by "deduplicate"? Do you mean some refactoring duplicated code (using this function in the pcntl code)?

Comment (bwoebi) [reply ext/ddtrace.c:2143]: Yes, refactoring the common code out into a single function.

**Review by bwoebi — APPROVED** [review]
Thanks, looks good :-)

**Pull request merged**